src/IterativeSolvers.fypp

```diff
@@ -550,22 +609,30 @@ contains
         integer :: i, k, niter
         ${type}$, allocatable :: alpha(:)
         class(abstract_vector_${type[0]}$${kind}$), allocatable :: dx, wrk
+        character(len=256) :: msg
 
         ! Deals with the optional args.
+        if (present(rtol)) then
+            rtol_ = rtol
+        else
+            rtol_ = rtol_${kind}$
+        end if
+        if (present(atol)) then
+            atol_ = atol
+        else
+            atol_ = atol_${kind}$
+        end if
```

Comment:
Couldn't we use `stdlib_optval` here to make the code a bit less verbose?

Reply:
done.